ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java

```diff
@@ -118,34 +132,38 @@ public void testBasicDeletion()
   @Test
   public void testBadCallback()
       throws Exception {
-    ServerErrorCode[] serverErrorCodes = new ServerErrorCode[9];
-    Arrays.fill(serverErrorCodes, ServerErrorCode.No_Error);
-    presetServerErrorCode(serverErrorCodes);
-    final CountDownLatch callbackCalled = new CountDownLatch(1);
-    List<Future> futures = new ArrayList<>();
-    for (int i = 0; i < 5; i++) {
-      if (i == 1) {
-        futures.add(router.deleteBlob(blobIdString, new Callback<Void>() {
+    testWithErrorCodes(Collections.singletonMap(ServerErrorCode.No_Error, 9), serverLayout, null,
+        new ErrorCodeChecker() {
           @Override
-          public void onCompletion(Void result, Exception exception) {
-            callbackCalled.countDown();
-            throw new RuntimeException("Throwing an exception in the user callback");
-          }
-        }));
-      } else {
-        futures.add(router.deleteBlob(blobIdString));
-      }
-    }
-    for (Future future : futures) {
-      future.get();
-    }
-    Assert.assertTrue("Callback not called.", callbackCalled.await(2, TimeUnit.SECONDS));
-    Assert.assertEquals("All operations should be finished.", 0, router.getOperationsCount());
-    Assert.assertTrue("Router should not be closed", router.isOpen());
+          public void check(RouterErrorCode expectedError)
+              throws Exception {
+            final CountDownLatch callbackCalled = new CountDownLatch(1);
+            List<Future> futures = new ArrayList<>();
+            for (int i = 0; i < 5; i++) {
+              if (i == 1) {
+                futures.add(router.deleteBlob(blobIdString, new Callback<Void>() {
+                  @Override
+                  public void onCompletion(Void result, Exception exception) {
+                    callbackCalled.countDown();
+                    throw new RuntimeException("Throwing an exception in the user callback");
+                  }
+                }));
+              } else {
+                futures.add(router.deleteBlob(blobIdString));
+              }
+            }
+            for (Future future : futures) {
+              future.get();
+            }
+            Assert.assertTrue("Callback not called.", callbackCalled.await(2, TimeUnit.SECONDS));
+            Assert.assertEquals("All operations should be finished.", 0, router.getOperationsCount());
+            Assert.assertTrue("Router should not be closed", router.isOpen());
 
-    //Test that DeleteManager is still operational
-    router.deleteBlob(blobIdString, new ClientCallback()).get();
-    Assert.assertTrue("Callback not called.", operationCompleteLatch.await(2, TimeUnit.SECONDS));
+            //Test that DeleteManager is still operational
+            router.deleteBlob(blobIdString, new ClientCallback()).get();
+            Assert.assertTrue("Callback not called.", operationCompleteLatch.await(2, TimeUnit.SECONDS));
```

Comment:
I know this is not part of your change but the way `operationCompleteLatch` is used, it opens up room for a lot of mistakes. Why not pass a countdown latch when the `ClientCallback` is created?